contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -332,6 +336,12 @@ abstract contract CompoundingValidatorManager is Governable, Pausable {
         deposits[validator.depositDataRoot] = DepositData({
             pubKeyHash: pubKeyHash,
             amountGwei: depositAmountGwei,
+            /// After the Pectra upgrade the validators have a new restriction when proposing 
+            /// blocks. The timestamps are at strict intervals of 12 seconds from the genesis block
+            /// forward. Each slot is created at strict 12 second intervals and those slots can
+            /// either have blocks attached to them or not. This way using the block.timestamp
+            /// the slot number can easily be calculated. 
+            slotNumber: SafeCast.toUint64((block.timestamp - GENESIS_TIME) / 12),
```

Comment:
We can do the conversion in verifyDeposits and verifyBalances rather than store it in storage